features/dd-sdk-android-rum/src/test/kotlin/com/datadog/android/rum/internal/domain/scope/RumApplicationScopeAttributePropagationTest.kt

```diff
@@ -0,0 +1,198 @@
+    private lateinit var mockInitialResourceIdentifier: InitialResourceIdentifier
+
+    lateinit var fakeEventTime: Time
+
+    lateinit var fakeEvent: RumRawEvent
+
+    @Forgery
+    lateinit var fakeTimeInfoAtScopeStart: TimeInfo
+
+    @Forgery
+    lateinit var fakeNetworkInfoAtScopeStart: NetworkInfo
+
+    @Forgery
+    lateinit var fakeDatadogContext: DatadogContext
+
+    @BoolForgery
+    var fakeHasReplay: Boolean = false
+
+    @FloatForgery(min = 0f, max = 100f)
+    var fakeSampleRate: Float = 0f
+
+    @BoolForgery
+    var fakeBackgroundTrackingEnabled: Boolean = false
+
+    @BoolForgery
+    var fakeTrackFrustrations: Boolean = true
+
+    @StringForgery
+    lateinit var fakeApplicationId: String
+
+    @BeforeEach
+    fun `set up`(forge: Forge) {
+        fakeDatadogContext = fakeDatadogContext.copy(
+            source = forge.aValueFrom(ViewEvent.ViewEventSource::class.java).toJson().asString
+        )
+
+        val fakeOffset = -forge.aLong(1000, 50000)
+        val fakeTimestamp = System.currentTimeMillis() + fakeOffset
+        val fakeNanos = System.nanoTime() + TimeUnit.MILLISECONDS.toNanos(fakeOffset)
+        val maxLimit = max(Long.MAX_VALUE - fakeTimestamp, Long.MAX_VALUE)
+        val minLimit = min(-fakeTimestamp, maxLimit)
+
+        fakeDatadogContext = fakeDatadogContext.copy(
+            time = fakeTimeInfoAtScopeStart.copy(
+                serverTimeOffsetMs = forge.aLong(min = minLimit, max = maxLimit)
+            )
+        )
+        fakeEventTime = Time(fakeTimestamp, fakeNanos)
+
+        whenever(mockRumFeatureScope.withWriteContext(any(), any())) doAnswer {
+            val callback = it.getArgument<(DatadogContext, EventBatchWriter) -> Unit>(1)
+            callback.invoke(fakeDatadogContext, mockEventBatchWriter)
+        }
+
+        whenever(mockSdkCore.internalLogger) doReturn mock()
+
+        testedScope = RumApplicationScope(
+            applicationId = fakeApplicationId,
+            sdkCore = mockSdkCore,
+            sampleRate = fakeSampleRate,
+            backgroundTrackingEnabled = fakeBackgroundTrackingEnabled,
+            trackFrustrations = fakeTrackFrustrations,
+            firstPartyHostHeaderTypeResolver = mockResolver,
+            cpuVitalMonitor = mockCpuVitalMonitor,
+            memoryVitalMonitor = mockMemoryVitalMonitor,
+            frameRateVitalMonitor = mockFrameRateVitalMonitor,
+            sessionEndedMetricDispatcher = mockSessionEndedMetricDispatcher,
+            sessionListener = mockSessionListener,
+            initialResourceIdentifier = mockInitialResourceIdentifier,
+            lastInteractionIdentifier = mockLastInteractionIdentifier
+        )
+    }
+
+    // region Propagate parent attributes
+
+    @Test
+    fun `M return global attributes W getCustomAttributes()`() {
+        // When
+        val customAttributes = testedScope.getCustomAttributes()
+
+        // Then
+        assertThat(customAttributes).isEmpty()
+    }
+
+    // TODO
```

Comment:
should it have a Jira ticket?

Reply:
That's a remnant that I didn't remove :D